### ISSUE TITLE
Why use clipnorm=0.001 for adam optimizer?

### ISSUE THREAD

**Issue by mailcorahul** [fizyr/keras-retinanet#942]
@hgaiser can you explain the reason for using clipnorm param with a value of 0.001 for adam?
Does this have anything to do with exploding gradient? In issue #515 , you have mentioned that you were facing exploding gradient problem with sgd and not with adam.
just curious to know. thanks.

https://github.com/fizyr/keras-retinanet/blob/c841da27f540084d27e971b6d00c178ff005d344/keras_retinanet/bin/train.py#L131

**Comment by hgaiser** [comment]
Yes, there were some experiments with different optimisers and Adam with this clip norm had the best result. As mentioned in the issue you referenced, I'm open to experiments with other optimisers.

**Comment by mailcorahul** [comment]
@hgaiser Thanks for the reply :-)
Also may I know the reason for using a very small learning rate(1e-5)? Was it because that lr did well for any particular benchmark dataset or for wide range of datasets? What was the benchmark dataset you used for fixing these hyperparameters?

**Comment by hgaiser** [comment]
@yhenon performed these experiments, I believe they were on COCO. I didn't spend much time to optimize those parameters.

**Comment by mailcorahul** [comment]
Thank you @hgaiser :-)

**Comment by guicho271828** [comment]
Just for a reference: https://arxiv.org/abs/1905.11881 
Why gradient clipping accelerates training: A theoretical justification for adaptivity
ICLR2021

**Comment by forddwork** [comment]
Generally for any SGD, it permits faster learning rate to be used (i.e., less careful choice of lr becomes ok). If some dimensions dont like that e.g., have steep 'cliffs', the gradient is clipped there to prevent the parameters from overstepping/jumping into places where the overall loss visibly spikes up suddenly during training. It can take several extra epochs to fix/come back to the loss value which was before the spike (and this may not be at a better spot than before), so taking that into account the overall learning process is more efficient (faster).
